middleware/index.ts

```diff
@@ -258,3 +258,72 @@ export default function applySfFxMiddleware(request: any, state: any, resultArgs
     // Function params
     return [event, context, logger];
 }
+
+function getUserFn() {
+    const functionPath = process.env.USER_FUNCTION_URI || '/workspace';
+    const pjsonPath = path.join(functionPath, 'package.json');
+    let main;
+    try {
+        main = require(pjsonPath).main;
+    } catch (e) {
+        console.log(e);
+        throw `Could not read package.json: ${e}`;
+    }
+    const mainPath = path.join(functionPath, main);
+    let mod;
+    try {
+        mod = require(mainPath);
+    } catch (e) {
+        console.log(e);
+        throw `Could not locate user function: ${e}`;
+    }
+    if (mod.__esModule && typeof mod.default === 'function') {
+        return mod.default;
+    }
+    return mod;
+}
+
+function createLogger(requestID: string): Logger {
+    const logger = new Logger('Evergreen Logger');
+    const level = process.env.DEBUG ? LoggerLevel.DEBUG : LoggerLevel.INFO;
+
+    logger.addStream({stream: process.stderr});
+    logger.setLevel(level);
+
+    if (requestID) {
+        logger.addField('request_id', requestID);
+    }
+
+    return logger;
+}
+
+const userFn = getUserFn();
+
+export default async function systemFn(message: object): Promise<object> {
+    const payload = message['payload'];
+
+    // Remap headers to a standard JS object
+    const headers = message['headers'].toRiffHeaders();
+    Object.keys(headers).map((key: string) => {headers[key] = message['headers'].getValue(key)});
+
+    const requestId = headers['ce-id'] || headers['x-request-id'];
+    const requestLogger = createLogger(requestId);
+    try {
+        const middlewareResult = await applySfFxMiddleware(
```

Comment:
I guess since systemFn & middleware are combined... so we no longer need to consider chaining multiple middleware fn?

Reply:
Do we still need the function builder (containing riff)?

Where is the riff invoker builder? I probably got things mixed up, I thought riff+systemFn was in the function builder.

Reply:
We still need the riff node-function-buildpack.  It's still part of the image, but I already removed our System function from it, and placed it into it's own separate buildpack before this PR.  My intention was to make sure everything still worked after removing the System function from the riff node-function-buildpack before making changes to combine the System and Middleware functions.